### DISCUSSION TITLE
Gradebook

### DISCUSSION THREAD

**Discussion by gbowne1** [gbowne1/gradebook-management-system#9] [Announcements]
Development on this Gradebook program has officially started.

Use this Discussions tab for any discussion related to Gradebook, but not specifically meant for any particular issue or bug report.

**Comment by gbowne1** [comment]
@andrewlod 

Have we decided on GPA?

I did some research on K-12 education and grading over the past few weeks .

I'm going to work on improving and debugging the files we have.

As far as [Test] we should use either Google test or CTest.

**Reply by andrewlod** [reply]
Hello!

By GPA do you mean the average grade calculation system?

I have a couple thoughts about grading:
- For flexible grading system, I think it would be quite challenging to cover all possible grading systems, but we could divide them into the following:
  - Number-based grading system, such as 0-10, where the user can specify min, max and a cutting point that divides pass/fail grades.
  - Letter-based grading system, such as A-F, where the user register each single letter as a grade and decides which ones are pass and which ones are fail.
- About average grading systems, I am not sure if letting the user input a formula would be a good idea. It would be flexible, but maybe too powerful and confusing. I suppose we can come up with some presets (basic average, weighted average, some other aggregation method...)
- When it comes to GPA, as far as I am aware, the range is always 0-4, right? I have seen some exceptions where it is 0-5 when AP classes are included

> As far as [Test] we should use either Google test or CTest.

Great! I have worked with Google Test before and I am familiar with it.

**Reply by gbowne1** [reply]
I'm primarily focused on K-12 grades but that doesnt mean it could not be used in higher grades or college and university or in private schools.

I also realized that gradebook.cpp might end up pretty large.  We can absolutely abstract the functions into classes, like Menu, etc.

I do like the idea of some commonly used presets.  That would maintain some sense of unity.

I have a chart I will include for GPA and also it includes the ranges and yes its 0.0 to 4.0

Great.  Do we include a menu system for running validation tests on the program?

I am familiar with both CTest and Google Test.

Use this discussion for any ideas discussion just to keep the issues tab relevant and clean. We can break down features into individual discussion threads.  Feel free to open issues and discussions.
